main.js

```diff
@@ -1,3 +1,60 @@
 import './style.css';
 
-console.log('Hello world');
+function enable_tooltip(tooltip, button) {
+    const buttonRect = button.getBoundingClientRect();
+    const top = buttonRect.top - buttonRect.height - 20;
+    const left = buttonRect.left;
+    tooltip.style.top = top + 'px';
+    tooltip.style.left = left + 'px';
+    tooltip.style.display = 'block';
+}
+
+function disable_tooltip(tooltip) {
+    tooltip.style.display = 'none';
+}
+
+const button_hover = document.getElementById("button-hover");
+const tooltip_hover = document.getElementById('tooltip-hover');
+
+button_hover.addEventListener('mouseover', () => {
+    enable_tooltip(tooltip_hover, button_hover);
+});
+
+button_hover.addEventListener('mouseout', () => {
+    disable_tooltip(tooltip_hover);
+});
+
+const button_click = document.getElementById('button-click');
+const tooltip_click = document.getElementById('tooltip-click');
+
+let pressed = false;
+document.addEventListener('click', (event) => {
+    if (pressed) {
+        disable_tooltip(tooltip_click);
+        pressed = false;
+    } else if (event.target === button_click) {
```

Comment:
If you add the <span> inside the button, as I have shown on lecture - it will be broken.

You can assign events to any count of buttons like so:

```javascript
function assignTooltip(button, tooltip) { // By function we can assign same behaviour for any count of buttons
  button.addEventListener('click', () => {
    enable_tooltip(button, tooltip)
  });

  function onClickOutside (event) {
    if (button.contains(event.target)) { // we check any deep target Node inside button, or button itself
      disable_tooltip(tooltip)
    }
    document.documentElement.removeEventListener('click', onClickOutside)
    // we remove global event when tooltip is closed
  }

  document.documentElement.addEventListener('click', onClickOutside) // we add global event only when tooltip is shown
}
```